lib/ClangImporter/ClangImporter.cpp

```diff
@@ -1018,16 +1018,36 @@ ClangImporter::create(ASTContext &ctx,
 
   // Set up the file manager.
   {
-    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS;
-    if (!ctx.SearchPathOpts.VFSOverlayFiles.empty() ||
-        importerOpts.ForceUseSwiftVirtualFileSystem) {
-      // If the clang instance has overlays it means the user has provided
-      // -ivfsoverlay options.  We're going to clobber their file system with
-      // the Swift file system, so warn about it.
-      if (!instance.getHeaderSearchOpts().VFSOverlayFiles.empty()) {
-        ctx.Diags.diagnose(SourceLoc(), diag::clang_vfs_overlay_is_ignored);
-      }
+    llvm::IntrusiveRefCntPtr<llvm::vfs::OverlayFileSystem> VFS;
```

Comment:
Instead of constructing VFS as below, could you ask clang to construct a VFS on top of the Swift VFS? There is a `createVFSFromCompilerInvocation` in `clang/Frontend/CompilerInstance.h` that you could use like this:
```
llvm::IntrusiveRefCntPtr<llvm::vfs::OverlayFileSystem> VFS =
  createVFSFromCompilerInvocation(instance.getInvocation(), instance.getDiagnostics(), ctx.SourceMgr.getFileSystem());
```

If that works, it'll simplify this logic a lot. And also it'll make it unnecessary to change FileSystem to OverlayFileSystem throughout Swift.

Reply:
That's a great idea, I didn't know about this function existed.